flow/layers/clip_path_layer.cc

```diff
@@ -21,21 +21,17 @@ void ClipPathLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
 
   SkRect previous_cull_rect = context->cull_rect;
   SkRect clip_path_bounds = clip_path_.getBounds();
-  children_inside_clip_ = context->cull_rect.intersect(clip_path_bounds);
-  if (children_inside_clip_) {
-    TRACE_EVENT_INSTANT0("flutter", "children inside clip rect");
-
-    Layer::AutoPrerollSaveLayerState save =
-        Layer::AutoPrerollSaveLayerState::Create(context, UsesSaveLayer());
-    context->mutators_stack.PushClipPath(clip_path_);
-    SkRect child_paint_bounds = SkRect::MakeEmpty();
-    PrerollChildren(context, matrix, &child_paint_bounds);
-
-    if (child_paint_bounds.intersect(clip_path_bounds)) {
-      set_paint_bounds(child_paint_bounds);
-    }
-    context->mutators_stack.Pop();
+  context->cull_rect.intersect(clip_path_bounds);
+  Layer::AutoPrerollSaveLayerState save =
+      Layer::AutoPrerollSaveLayerState::Create(context, UsesSaveLayer());
+  context->mutators_stack.PushClipPath(clip_path_);
+  SkRect child_paint_bounds = SkRect::MakeEmpty();
+  PrerollChildren(context, matrix, &child_paint_bounds);
+
+  if (child_paint_bounds.intersect(clip_path_bounds)) {
+    set_paint_bounds(child_paint_bounds);
```

Comment:
It occurs to me that I should set this bounds unconditionally now so that it doesn't retain the "unvisited" bounds value. I'm not sure what value this conditional had anyway as it is just avoiding storing "empty" bounds over other "empty" bounds - 4 float stores. The savings was not likely that great.

Reply:
In looking through it again, this code should be fine. The operands in the intersect operation are the bounds of the clip set here, which cannot change without constructing a new clip layer, and the bounds of the children, which cannot change without constructing a new clip layer to take the new list of children.